### PR TITLE
Fix string comparison in C++ UE4 client generator

### PR DESCRIPTION
**Pull request by wing328**
- Fix string comparison in C++ UE4 client generator

CC @ravinikam (2017/07) @stkrwork (2017/07) @etherealjoy (2018/02) @martindelille (2018/03) @muttleyxd (2019/08)

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

cc @Kahncode as well - the author of the generator

### CODE REVIEW

File: samples/client/petstore/cpp-ue4/Private/OpenAPIPetApiOperations.cpp [hunk 1]
```diff
@@ -220,7 +220,6 @@ void OpenAPIPetApi::FindPetsByStatusResponse::SetHttpResponseCode(EHttpResponseC
 	switch ((int)InHttpResponseCode)
 	{
 	case 200:
-	default:
```
Comment (Kahncode) [inline samples/client/petstore/cpp-ue4/Private/OpenAPIPetApiOperations.cpp:223]: Has something else changed? These default clauses are needed if I remember correctly.

Comment (wing328) [reply samples/client/petstore/cpp-ue4/Private/OpenAPIPetApiOperations.cpp:223]: Yes, there was a fix. Previously the first response code is considered the `default`, which is incorrect as `default` should be defined separately.

Shall we revise the code to throw an exception instead as the response code is not defined in the spec?

Comment (Kahncode) [reply samples/client/petstore/cpp-ue4/Private/OpenAPIPetApiOperations.cpp:223]: No, this code does not support exceptions. Essentially if the response code was undefined in the spec I assumed it was a correct response. You may change that default behavior but I believe my tests ended up concluding that this was the best case scenario.

Comment (wing328) [reply samples/client/petstore/cpp-ue4/Private/OpenAPIPetApiOperations.cpp:223]: What about logging an error message instead?

Comment (wing328) [reply samples/client/petstore/cpp-ue4/Private/OpenAPIPetApiOperations.cpp:223]: I think I'll merge this PR which aims to fix the string comparison. 

The issue (default case has gone) is present in the generator regardless of this PR.

**Pull request merged**